File: gma-next-gen-sdk-android-examples-main/java/nextgenexample/app/src/main/java/com/example/nextgenexample/iconad/IconAdFragment.java
/*
 * Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.example.nextgenexample.iconad;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.example.nextgenexample.AdFragment;
import com.example.nextgenexample.Constant;
import com.example.nextgenexample.databinding.FragmentIconBinding;
import com.example.nextgenexample.databinding.IconAdBinding;
import com.google.android.libraries.ads.mobile.sdk.common.AdChoicesPlacement;
import com.google.android.libraries.ads.mobile.sdk.common.AdLoadCallback;
import com.google.android.libraries.ads.mobile.sdk.common.AdValue;
import com.google.android.libraries.ads.mobile.sdk.common.FullScreenContentError;
import com.google.android.libraries.ads.mobile.sdk.common.LoadAdError;
import com.google.android.libraries.ads.mobile.sdk.iconad.IconAd;
import com.google.android.libraries.ads.mobile.sdk.iconad.IconAdEventCallback;
import com.google.android.libraries.ads.mobile.sdk.iconad.IconAdPlacement;
import com.google.android.libraries.ads.mobile.sdk.iconad.IconAdRequest;
import com.google.android.libraries.ads.mobile.sdk.iconad.IconAdView;

/** An [AdFragment] subclass that loads an icon ad. */
public class IconAdFragment extends AdFragment<FragmentIconBinding> {
  // Default constructor required for fragment instantiation.
  public IconAdFragment() {}

  // Sample icon ad unit ID.
  private static final String AD_UNIT_ID = "ca-app-pub-3940256099942544/1476272466";

  private IconAd iconAd;

  @Override
  protected BindingInflater<FragmentIconBinding> getBindingInflater() {
    return com.example.nextgenexample.databinding.FragmentIconBinding::inflate;
  }

  @Override
  public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
    super.onViewCreated(view, savedInstanceState);
    loadIconAd();
  }

  @Override
  public void onDestroyView() {
      super.onDestroyView();

      // Always call destroy() on ads on removal.
      if (iconAd != null) {
          iconAd.destroy();
          iconAd = null;
      }
  }

  // [START load_ad]
  private void loadIconAd() {
    IconAdRequest request =
        new IconAdRequest.Builder(AD_UNIT_ID)
            // The "AdChoices" badge is rendered at the top right corner of the icon ad
            // if left unspecified.
            .setAdChoicesPlacement(AdChoicesPlacement.BOTTOM_RIGHT)
            // It is recommended to specify the placement of your icon ad
            // to help Google optimize your icon ad performance.
            .setIconAdPlacement(IconAdPlacement.BROWSER)
            .build();

    IconAd.load(
        request,
        new AdLoadCallback<IconAd>() {
          @Override
          public void onAdFailedToLoad(@NonNull LoadAdError adError) {
            Log.w(Constant.TAG, "Icon ad failed to load :" + adError);
            showToast("Icon ad failed to load with error code: " + adError.getCode());
          }

          @Override
          public void onAdLoaded(@NonNull IconAd ad) {
            Log.d(Constant.TAG, "Icon ad loaded.");

            // Always call destroy() on ads on removal.
            if (iconAd != null) {
                iconAd.destroy();
            }
            iconAd = ad;
            setAdEventCallback(ad);
            displayIconAd(ad);
          }
        });
  }

  // [END load_ad]

  private void setAdEventCallback(IconAd iconAd) {
    // [START ad_events]
    iconAd.setAdEventCallback(
        new IconAdEventCallback() {
          @Override
          public void onAdShowedFullScreenContent() {
            // Icon ad showed full screen content.
          }

          @Override
          public void onAdDismissedFullScreenContent() {
            // Icon ad dismissed full screen content.
          }

          @Override
          public void onAdFailedToShowFullScreenContent(
              @NonNull FullScreenContentError fullScreenContentError) {
            // Icon ad failed to show full screen content.
          }

          @Override
          public void onAdImpression() {
            // Icon ad recorded an impression.
          }

          @Override
          public void onAdClicked() {
            // Icon ad recorded a click.
          }

          @Override
          public void onAdPaid(@NonNull AdValue value) {
            // Icon ad estimated to have earned money.
          }
        });
    // [END ad_events]
  }

  private void displayIconAd(IconAd iconAd) {
    if (getActivity() == null) {
      return;
    }

    getActivity()
        .runOnUiThread(
            () -> {
              // [START populate_ad]
              IconAdBinding iconAdViewBinding = IconAdBinding.inflate(getLayoutInflater());
              // Add the ad view to the active view hierarchy.
              binding.iconAdContainer.addView(iconAdViewBinding.getRoot());
              IconAdView iconAdView = iconAdViewBinding.getRoot();

              // Populate the view elements with their respective icon ad asset.
              iconAdView.setCallToActionView(iconAdViewBinding.adCallToAction);
              iconAdView.setHeadlineView(iconAdViewBinding.adHeadline);
              iconAdView.setIconView(iconAdViewBinding.adIcon);
              iconAdView.setStarRatingView(iconAdViewBinding.adStars);
              // [END populate_ad]

              // [START register_ad]
              // Map each asset view property to the corresponding view in your view hierarchy.
              iconAdViewBinding.adCallToAction.setText(iconAd.getCallToAction());
              iconAdViewBinding.adHeadline.setText(iconAd.getHeadline());
              iconAdViewBinding.adIcon.setImageDrawable(iconAd.getIcon().getDrawable());

              if (iconAd.getStarRating() != null) {
                iconAdViewBinding.adStars.setRating(iconAd.getStarRating().floatValue());
              }

              // Register the icon ad with the view presenting it.
              iconAdView.registerIconAd(iconAd);
              // [END register_ad]
            });
  }
}

File: gma-next-gen-sdk-android-examples-main/kotlin/nextgenexample/app/src/main/java/com/example/next_gen_example/icon/IconAdFragment.kt
/*
 * Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.example.next_gen_example.icon

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.next_gen_example.AdFragment
import com.example.next_gen_example.Constant
import com.example.next_gen_example.databinding.FragmentIconAdBinding
import com.example.next_gen_example.databinding.IconAdBinding
import com.google.android.libraries.ads.mobile.sdk.common.AdChoicesPlacement
import com.google.android.libraries.ads.mobile.sdk.common.AdLoadCallback
import com.google.android.libraries.ads.mobile.sdk.common.AdValue
import com.google.android.libraries.ads.mobile.sdk.common.FullScreenContentError
import com.google.android.libraries.ads.mobile.sdk.common.LoadAdError
import com.google.android.libraries.ads.mobile.sdk.iconad.IconAd
import com.google.android.libraries.ads.mobile.sdk.iconad.IconAdEventCallback
import com.google.android.libraries.ads.mobile.sdk.iconad.IconAdPlacement
import com.google.android.libraries.ads.mobile.sdk.iconad.IconAdRequest

class IconAdFragment : AdFragment<FragmentIconAdBinding>() {
  override val bindingInflater: (LayoutInflater, ViewGroup?, Boolean) -> FragmentIconAdBinding
    get() = FragmentIconAdBinding::inflate

  private var iconAd: IconAd? = null

  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)
    loadIconAd()
  }

  override fun onDestroyView() {
    super.onDestroyView()
    // Always call destroy() on ads on removal.
    iconAd?.destroy()
    iconAd = null
  }

  // [START load_ad]
  private fun loadIconAd() {
    val request =
      IconAdRequest.Builder(AD_UNIT_ID)
        // The "AdChoices" badge is rendered at the top right corner of the icon ad
        // if left unspecified.
        .setAdChoicesPlacement(AdChoicesPlacement.BOTTOM_RIGHT)
        // It is recommended to specify the placement of your icon ad
        // to help Google optimize your icon ad performance.
        .setIconAdPlacement(IconAdPlacement.BROWSER)
        .build()

    IconAd.load(
      request,
      object : AdLoadCallback<IconAd> {
        override fun onAdFailedToLoad(adError: LoadAdError) {
          Log.w(Constant.TAG, "Icon ad failed to load: $adError")
          showToast("Icon ad failed to load.")
        }

        override fun onAdLoaded(ad: IconAd) {
          Log.d(Constant.TAG, "Icon ad loaded")
          // Always call destroy() on ads on removal.
          iconAd?.destroy()
          iconAd = ad
          setAdEventCallback(ad)
          displayIconAd(ad)
        }
      },
    )
  }

  // [END load_ad]

  private fun setAdEventCallback(iconAd: IconAd) {
    // [START ad_events]
    iconAd.adEventCallback =
      object : IconAdEventCallback {
        override fun onAdShowedFullScreenContent() {
          // Icon ad showed full screen content.
        }

        override fun onAdDismissedFullScreenContent() {
          // Icon ad dismissed full screen content.
        }

        override fun onAdFailedToShowFullScreenContent(
          fullScreenContentError: FullScreenContentError
        ) {
          // Icon ad failed to show full screen content.
        }

        override fun onAdImpression() {
          // Icon ad recorded an impression.
        }

        override fun onAdClicked() {
          // Icon ad recorded a click.
        }

        override fun onAdPaid(value: AdValue) {
          // Icon ad estimated to have earned money.
        }
      }
    // [END ad_events]
  }

  private fun displayIconAd(iconAd: IconAd) {
    activity?.runOnUiThread {
      // [START populate_ad]
      val iconAdViewBinding = IconAdBinding.inflate(layoutInflater)
      // Add the ad view to the active view hierarchy.
      binding.iconAdContainer.addView(iconAdViewBinding.root)
      val iconAdView = iconAdViewBinding.root

      // Populate the view elements with their respective icon ad asset.
      iconAdView.callToActionView = iconAdViewBinding.adCallToAction
      iconAdView.headlineView = iconAdViewBinding.adHeadline
      iconAdView.iconView = iconAdViewBinding.adIcon
      iconAdView.starRatingView = iconAdViewBinding.adStars
      // [END populate_ad]

      // [START register_ad]
      // Map each asset view property to the corresponding view in your view hierarchy.
      iconAdViewBinding.adCallToAction.text = iconAd.callToAction
      iconAdViewBinding.adHeadline.text = iconAd.headline
      iconAdViewBinding.adIcon.setImageDrawable(iconAd.icon.drawable)
      iconAd.starRating?.toFloat().also { value ->
        if (value != null) {
          iconAdViewBinding.adStars.rating = value
        }
      }

      // Register the icon ad with the view presenting it.
      iconAdView.registerIconAd(iconAd)
      // [END register_ad]
    }
  }

  private companion object {
    const val AD_UNIT_ID = "ca-app-pub-3940256099942544/1476272466"
  }
}
